ddtrace/tracer.py

```diff
@@ -608,24 +611,16 @@ def write(self, spans):
         Send the trace to the writer to enqueue the spans list in the agent
         sending queue.
         """
-        if not spans:
+        if not spans or not self.enabled:
```

Comment:
This looks unrelated to the purpose of this patch, no?

Reply:
Hm, it was already being done previously but it was done later. I'll move it back to where it was originally.